packages/block-editor/src/store/selectors.js

```diff
@@ -460,6 +460,21 @@ export const getBlockParents = createSelector(
 	( state ) => [ state.blocks.parents ]
 );
 
+/**
+ * Given a block client ID, returns the direct parent.
+ *
+ * @param {Object} state    Editor state.
+ * @param {string} clientId Block from which to find root client ID.
+ *
+ * @return {string} ClientID of the parent block.
+ */
+export const getBlockParent = createSelector(
+	( state, clientId ) => {
+		return state.blocks.parents[ clientId ];
+	},
+	( state ) => [ state.blocks.parents ]
```

Comment:
Not totally sure, maybe we should change this to `( state, clientId ) => [ state.blocks.parents[ clientID ] ]` so that it only refreshes on changes to its own parent, not any parent